modules/docs/arrow-docs/docs/docs/patterns/glossary/README.md

```diff
@@ -24,6 +24,21 @@ branching in code with [`Either`]({{ '/docs/datatypes/either' | relative_url }})
 catching exceptions with [`Try`]({{ '/docs/datatypes/try' | relative_url }}),
 or interacting with the platform the program runs in using [`IO`]({{ '/docs/effects/io' | relative_url }}).
 
+The name datatype comes from the representation of these patterns using plain data, with `data` and `sealed` classes.
+For example, the internal representation of an `Option` is a sealed class with two data classes `Some<A>(val a: A)` and `None`,
+and `Ior` is a sealed class with three data class inheritors, `Left(val a: A)`, `Right(val b: B)`, and `Both(val a: A, val b: B)`.
+
+Lazy datatypes can be recursive, nesting instances of themselves until execution. One example is `IO`.
```

Comment:
For some reason I feel like there's something missing before this paragraph. You talk about lazy data types and I'm not sure if reader would not what you're talking about exactly. I could imagine what is it, but I wouldn't expect devs not familiarized with FP to know. I would at least introduce laziness around data types before or link to it.